src/main/java/ch/njol/skript/expressions/ExprSpawnEggEntity.java

```diff
@@ -0,0 +1,123 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.ItemUtils;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.slot.Slot;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntitySnapshot;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.SpawnEggMeta;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Spawn Egg Entity")
+@Description({
+	"Gets or sets the entity snapshot that the provided spawn eggs will spawn when used.",
+	"NOTE: When using an entity data i.e 'a zombie', it will spawn a new zombie to retrieve an entity snapshot and then be removed."
+})
+@Examples({
+	"set {_item} to a zombie spawn egg",
+	"broadcast the spawn egg entity of {_item}",
+	"",
+	"spawn a pig at location(0,0,0):",
+		"\tset the max health of entity to 20",
+		"\tset the health of entity to 20",
+		"\tset {_snapshot} to the entity snapshot of entity",
+		"\tclear entity",
+	"set the spawn egg entity of {_item} to {_snapshot}",
+	"if the spawn egg entity of {_item} is {_snapshot}: # Minecraft 1.20.5+",
+	"",
+	"set the spawn egg entity of {_item} to (random element out of all entities)",
+	"",
+	"set the spawn egg entity of {_item} to a zombie"
+})
+@RequiredPlugins("Minecraft 1.20.2+, Minecraft 1.20.5+ (comparisons)")
+@Since("INSERT VERSION")
+public class ExprSpawnEggEntity extends SimplePropertyExpression<Object, Object> {
+
+	static {
+		if (Skript.classExists("org.bukkit.entity.EntitySnapshot"))
+			register(ExprSpawnEggEntity.class, Object.class, "spawn egg entity", "itemstacks/itemtypes/slots");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (Player.class.isAssignableFrom(exprs[0].getReturnType())) {
+			Skript.error("You can't set the spawn egg entity to a player.");
+			return false;
+		}
```

Comment:
exprs[0] refers to the `itemstacks/itemtypes/slots` part of the pattern here right? I dont think the `init` method is the place for this error anyway, looks like it belongs in the `change` method when runtime errors are implemented